pkg/sql/txn_instrumentation.go

```diff
@@ -128,21 +164,17 @@ func (h *txnInstrumentationHelper) StartDiagnostics(
 	h.diagnosticsCollector = newTxnDiagnosticsCollector(txnRequest, reqID, span)
 }
 
-// DiagnosticsInProgress returns true if diagnostics collection is considered
-// to be progress. This is determined by the following:
-// - there is an active request id
-// - there has been at least 1 statement bundle collected
-// - there are still statements to be collected
-// If none of these are true, then diagnostics collection is not in progress.
-//
-// NB: Currently, transaction diagnostics collection only happens through
-// requests, so the first condition is sufficient. If, in the future, new
-// methods of collecting transaction diagnostics are added, such as through
-// "explain analyze (debug)", then the other two conditions will be necessary.
+func (h *txnInstrumentationHelper) ReadyToStartDiagnostics() bool {
+	return h.diagnosticsCollector.state == txnDiagnosticsNotStarted
+}
+
 func (h *txnInstrumentationHelper) DiagnosticsInProgress() bool {
-	return h.diagnosticsCollector.requestId != 0 ||
-		len(h.diagnosticsCollector.stmtBundles) > 0 ||
-		len(h.diagnosticsCollector.stmtsFpsToCapture) > 0
+	return h.diagnosticsCollector.InProgress()
+}
+
+func (h *txnInstrumentationHelper) Abort() {
+	h.Reset()
+	h.diagnosticsCollector.UpdateState(txnDiagnosticsAborted)
```

Comment:
can you put a docstring about why it's ok to leave the collector in Aborted state after the instrumentation helper is reset. my assumption is that it could get re-used after this, right?

Reply:
or is it possible to collapse the "aborted" state into "not started"?